### ISSUE TITLE
UnusedPrivateField false positive

### ISSUE THREAD

**Issue by 1ma** [phpmd/phpmd#324]
I'm experiencing a false positive with the latest stable version of PHPMD.

Steps to reproduce:
1. clone this repository: https://github.com/MarcelHernandez/yny.git
2. cd into the root directory and run `phpmd src/ xml unusedcode`
3. output:

``` xml
<?xml version="1.0" encoding="UTF-8" ?>
<pmd version="@project.version@" timestamp="2015-11-07T21:58:50+01:00">
  <file name="/var/lib/jenkins/workspace/yny/src/AppBundle/Controller/FPController.php">
    <violation beginline="22" endline="22" rule="UnusedPrivateField" ruleset="Unused Code Rules" externalInfoUrl="http://phpmd.org/rules/unusedcode.html#unusedprivatefield" priority="3">
      Avoid unused private fields such as '$accountGateway'.
    </violation>
  </file>
</pmd>
```

Here is the source file that triggers the error, for quick reference:

``` php
<?php

namespace AppBundle\Controller;

use AppBundle\Gateway\AccountGateway;
use AppBundle\Gateway\TransactionGateway;
use JMS\DiExtraBundle\Annotation\Inject;
use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
use Symfony\Bundle\FrameworkBundle\Controller\Controller;

class FPController extends Controller
{
    /**
     * @var AccountGateway
     *
     * @Inject("gateway.account")
     */
    private $accountGateway;

    /**
     * @var TransactionGateway
     *
     * @Inject("gateway.transaction")
     */
    private $transactionGateway;

    /**
     * @Method("GET")
     * @Route("/testfp/{slug}", name="testfp")
     */
    public function testAction($slug)
    {
        $accountId = $this->accountGateway
            ->findBySlug($slug)['id'];

        $this->transactionGateway
            ->addTransaction($accountId, 'foo', 1234, '2015-11-07');

        return 'Fine.';
    }
}
```

Further details:

```
$ uname -a
Linux debian-wheezy 3.2.0-4-amd64 #1 SMP Debian 3.2.68-1+deb7u5 x86_64 GNU/Linux

$ php -v
PHP 5.5.30-1~dotdeb+7.1 (cli) (built: Oct  1 2015 17:38:39) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies
    with Xdebug v2.3.3, Copyright (c) 2002-2015, by Derick Rethans

$ phpmd --version
PHPMD 2.3.2
```
